front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -40,13 +51,51 @@ const useGetProjectedTrainOperationalPoints = (
         };
         const { operational_points } = await postPathProperties(pathPropertiesParams).unwrap();
 
-        setOperationalPoints(
-          operational_points as NonNullable<PathProperties['operational_points']>
-        );
+        let operationalPointsWithAllWaypoints = operational_points!;
+
+        // Check if there are vias added by map click and insert them in the operational points
+        let waypointCounter = 1;
+
+        trainScheduleUsedForProjection.path.forEach((step, i) => {
+          if (!('track' in step)) return;
+
+          const positionOnPath = pathfindingResult.path_item_positions[i];
+          const indexToInsert = operationalPointsWithAllWaypoints.findIndex(
+            (op) => op.position >= positionOnPath
+          );
+
+          const formattedStep: NonNullable<PathProperties['operational_points']>[number] = {
+            id: step.id,
+            extensions: {
+              identifier: {
+                name: t('requestedPoint', { count: waypointCounter }),
+                uic: 0,
+              },
+            },
+            part: { track: step.track, position: step.offset },
+            position: positionOnPath,
+          };
+
+          // If we can't find any op position greater than the current step position, we add it at the end
+          if (indexToInsert === -1) {
+            operationalPointsWithAllWaypoints.push(formattedStep);
+            return;
+          }
+
+          operationalPointsWithAllWaypoints = addElementAtIndex(
+            operationalPointsWithAllWaypoints,
+            indexToInsert,
+            formattedStep
+          );
+
+          waypointCounter += 1;
```

Comment:
Something was bugging me about this code, originally I thought it was inconsequential so ignored it but during a shower I realized it might matter after all…

What if the last two waypoints are track offsets? In that case, we'd take the `push()` route, but we still want the `waypointCounter` to be incremented to avoid having these share the same number when presented to the user. So I think we should move this line up before `if (indexToInsert === -1)`?

Reply:
Wow very nice shower inspiration, you were right ! Fixed :)